src/Components/GiftList/Gift/Gift.js

```diff
@@ -0,0 +1,80 @@
+import React from 'react';
+import './Gift.scss';
+
+class Gift extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      gift: [],
+      btnActive: false,
+      btnAddCart: false,
+    };
+  }
+  mouseIn = () => {
+    this.setState({
+      btnActive: false,
+    });
+  };
+
+  mouseOut = () => {
+    this.setState({
+      btnActive: true,
+    });
+  };
+
+  addCart = () => {
+    this.setState({ btnAddCart: true });
+  };
+
+  componentDidMount() {
+    this.setState({
+      gift: this.props,
+      btnActive: this.props,
+    });
+  }
```

Comment:
```suggestion
  componentDidMount() {
    this.setState({
      gift: this.props.gift,
    });
  }
```
이런식으로 받아오시면 되는데, btnActive는 초기값이 false로 되어있으니, 굳이 setState를 할 필요는 없지 않을까요?

Reply:
DidMount에서 setState로 값을 전달해주지않으면 컴포넌트가 처음에 렌더링 되었을때 버튼이 보이는 상태로 뜨더라구요..
원래 이솝페이지에서는 처음 페이지가 렌더링되었을때는 안보이다가 마우스를 포인팅했을시에 버튼이 보이는방식이여서요..🧐

Reply:
아! 최상단 constructor생성자에서 boolean 값을 바꿔주고
didmount에서 없애니까 잘되네요 감사합니다!